xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-icon/plugin.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+(function () {
+  'use strict';
+
+  CKEDITOR.plugins.add('xwiki-icon', {
+    init: function (editor) {
+      editor.config.mentions = editor.config.mentions || [];
+      // Icon Autocomplete
+      editor.config.mentions.push({
+        marker: `/icon::`,
+        pattern: new RegExp('/icon::\\S{0,30}$'),
+        minChars: 0,
+        itemsLimit: 0,
+        itemTemplate: '<li data-id="{id}">' +
+          '<div class="ckeditor-autocomplete-item-head">' +
+          '<span class="ckeditor-autocomplete-item-icon-wrapper">' +
+          '<span class="{iconClass}" aria-hidden="true"></span>' +
+          '<img src="{imgSrc}" aria-hidden="true"></img>' +
```

Comment:
From my understanding, aria-hidden should be here when a screen reader would see duplicated content, in this case, we have the icon followed by its name.
@Sereza7, am I approaching this correctly, should I implement this differently?

Reply:
So, first: you shouldn't use aria-hidden both on the image and the span describing it, because you don't have anything to read anymore (no problem of duplicate, but nothing is read out, that's no good). I recommend to hide the image and leave the span as is.
Let me know if I misunderstood it, I didn't dig deep to check what those are used for exactly.

___

Then: Check out the HTML code style I updated after a proposal a few days ago ;)
https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XhtmlCssCodeStyle/

Seems like it's a case of "visually showed, semantically hidden" element.
I recommend to use `role='presentation'` on the image.
Using 'aria-hidden' is not wrong, but for the sake of consistency in xwiki-platform you should use the presentation role instead. They are [very similar solutions](https://timwright.org/blog/2016/11/19/difference-rolepresentation-aria-hiddentrue/), and in this case, they behave the same.

Reply:
The span is not describing the image, it's here to hold the equivalent icon when it is rendered using a css class, and can be empty when an image rendered icon is used instead. The describing span is later in the template.

Thank you for the documentation link, I'm checking it out :)
I'll use the `role='presentation'` on the image and add a comment.

Thanks.